### ISSUE TITLE
Separating JSON output to break up multiline messages into objects

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#1395]
**Is your feature request related to a problem? Please describe.**
The JSON output from `--verifytypes` is just a list of messages.  A multiline message is one big string, which is hard to analyze.

**Describe the solution you'd like**
A more object-oriented format.

**Additional context**
Example.  If you run it on pandas (by creating a `py.typed` file and renaming `tests`), you get an item like this:
```json
 {
            "file": "",
            "severity": "error",
            "message": "Type partially unknown for class \"pandas.compat.numpy.function.CompatValidator\"\n  Type partially unknown for class \"CompatValidator\"\n    Type partially unknown for method \"__init__\"\n      Parameter \"defaults\" is missing a type annotation\n      Parameter \"fname\" is missing a type annotation\n      Parameter \"max_fname_arg_count\" is missing a type annotation\n    Type partially unknown for method \"__call__\"\n      Parameter \"args\" is missing a type annotation\n      Parameter \"kwargs\" is missing a type annotation\n      Parameter \"fname\" is missing a type annotation\n      Parameter \"max_fname_arg_count\" is missing a type annotation\n    Type unknown for variable \"fname\"\n    Type unknown for variable \"method\"\n    Type unknown for variable \"defaults\"\n    Type unknown for variable \"max_fname_arg_count\""
        },
```
Would be nicer if it was something like:
```json
{
error_type: "Type partially unknown",
class: "pandas.compat.numpy.function.CompatValidator",
alias: "CompatValidator",
methods: [
   {
   name: "__init__",
   parameters: [
       {
       name: "defaults",
       error:  "Missing type annotation"
       },
      {
      name:  "fname",
      error: "Missing type annotation"
       },
      {
      name: "max_fname_arg_count",
      error:  "Missing type annotation"
      }
  ],
  },
  {
  name: "__call__",
  parameters: [
     {
     name: "args",
     error:  "Missing type annotation"
     },
// etcera
```

Benefit would be that you could then analyze the JSON object and look for errors that are common across different classes and methods, meaning that if you fix one, you might get rid of a lot of problems.

**Comment by erictraut** [comment]
I've started a new discussion about a rewrite of the verifytypes functionality here: https://github.com/microsoft/pyright/discussions/1568. I'd appreciate any thoughts you have.

**Comment by erictraut** [comment]
Thanks for your patience on this one. It took me longer than expected and required a complete rewrite of the type verifier.

The new version includes the following improvements:
1. The report eliminates a bunch of redundancy in its output.
2. As requested above, errors are much more granular rather than being reported as long, multi-line strings.
3. The report now distinguishes between "symbols that are exported by the package" and "symbols that are referenced but not directly exported". The latter includes types that are imported from other packages or private types that are not directly exported.
4. By default, missing docstring warnings are omitted from the text report, but they are included if you specify the "--verbose" option.
5. By default, symbols that have no errors are omitted from the text report, but they are included if you specify the "--verbose" option. They are always included in the json-based report.
6. All symbols include "reference counts" in the report to make it easier to tell which symbols are the "primary culprits". The text report prints a reference count only if it's greater than 1 for a symbol. The reference count (regardless of its value) is always included in the json output.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.126, which I just published.
